src/adsp_config.c

```diff
@@ -2337,15 +2337,50 @@ static int adsp_parse_config_fd(FILE *fd, struct image *image)
 	return ret;
 }
 
+static void copy_file(FILE *src, FILE *dst)
+{
+	char ch;
+	while ((ch = fgetc(src)) != EOF)
+		fputc(ch, dst);
+}
+
+static FILE* adsp_conf_files_open(const struct adsp_conf_files *files, const char *mode)
+{
+	FILE *file, *out_file;
+	int i;
+
+	out_file = fopen(files->out_file, "w");
+	if (!out_file) {
+		file_error("unable to open file for writing", files->out_file);
+		return NULL;
+	}
+
+	/* Merge multiple input toml files into one */
+	for (i = 0; i < files->file_count; i++) {
+		fprintf(out_file, "## File path: %s\n", files->file[i]);
+		file = fopen(files->file[i], "r");
+		if (!file) {
+			file_error("unable to open file for reading", files->file[i]);
+			fclose(out_file);
+			return NULL;
+		}
+		copy_file(file, out_file);
```

Comment:
use file pointer as parameter is not always proposed, directly read and write is also very simple.
btw, how about fread/fwrite? since toml file is not big.